Lib/contextlib.py

```diff
@@ -123,6 +128,7 @@ def __exit__(self, type, value, traceback):
                 #
                 if sys.exc_info()[1] is not value:
                     raise
+                return None
```

Comment:
I like @serhiy-storchaka's suggestion of making this more consistent with the `except RuntimeError` case, and making this clause be:

    # Use sys.exc_info() for consistency with the contextlib2 backport
    if sys.exc_info()[1] is value:
        return False
    raise

That way all the exception handling clauses are either `return <boolean expression>` (if suppressing the exception is one of the available options), or else a series of conditional `return False` statements followed by an unconditional `raise`.